### ISSUE TITLE
[CSS-images\ Specify fallback behavior when replaced or background image content not available

### ISSUE THREAD

**Issue by chrishtr** [w3c/csswg-drafts#1984]
There appears to be no spec language for what paints to the screen when the replaced content
of an <img> or CSS backgroud image is not available at paint time, either due to being in process
of being loaded, or due to an async decode.

Image fallbacks appear to be (partially?) specified in the presence of the image() function as part of
css images level 4. https://www.w3.org/TR/css-images-4/#image-fallbacks. This needs to be
extended more generally.

I think the best behavior is to fall back to whatever content painted underneath the paint phase that
would have drawn the image. e.g. if step 1.2. or step 7.1 of the algorithm below had no image content,
still paint but just skip those steps.
https://www.w3.org/TR/CSS2/zindex.html

See also https://github.com/whatwg/html/issues/1920

@vmpstr @tabatkins

**Comment by fantasai** [comment]
See “invalid image” in https://drafts.csswg.org/css-images-3/#image-values which is the spec prose that's currently active (no one implements `image()` yet). I suppose we can extend the wording to include images that have not yet loaded.

**Comment by tabatkins** [comment]
Yeah, I think "invalid image" should cover the loading case as well.

**Comment by AmeliaBR** [comment]
For content images, at least, browsers treat loading images (blank space or partial image render) differently from invalid images (broken-image icon).

Not sure whether its worthwhile to have a similar distinction for CSS images. It might be worth testing to see what browsers currently do.

**Comment by tabatkins** [comment]
Yeah, letting the UA discriminate between those cases when doing its ultimate fallback seems fine, and will just need some extra prose. But it would be good, I think, to let `image()` fallback handle loading images as well.

(`image()` will need to discriminate here, too - we don't want a loading fallback to fall back to another URL! Basically, if it's currently depending on a loading image, it should skip all further images that require loading, and fall back all the way to a CSS-generated image or color.)

**Comment by fantasai** [comment]
OK, we've defined “loading image” as a term somewhat distinct from “invalid image”, see changes at: https://github.com/w3c/csswg-drafts/commit/846d21d922fa4a47abea7be3ab57bcf3ebf62395

Agenda+ for CSSWG review.

**Comment by henke37** [comment]
jonathanKingston/css-fetch-integration#1 seems relevant when it comes to defining how images are loaded.

**Comment by litherum** [comment]
The current text looks fine.

**Comment by emilio** [comment]
FWIW we have some edge cases in this area that would be nice to clarify, including https://github.com/w3c/csswg-drafts/issues/218 / https://github.com/w3c/csswg-drafts/issues/2831, where there's no interop in sight, but they may not be in scope of this issue.
